### ISSUE TITLE
`kv.set()` doesn't work with four arguments

### ISSUE THREAD

**Issue by jwalton** [silas/node-consul#80]
All of the sudden I started running into problems where, when I tried to `set()` a KV, the key that was actually getting set was '[object Object]`.

After a lot of debugging, I tracked it to here: https://github.com/silas/node-consul/blob/master/lib/kv.js#L111-L119

```js
Kv.prototype.set = function(opts, callback) {
  switch (arguments.length) {
    case 4:
      // set(key, value, opts, callback)
      opts = arguments[2];
      opts.key = arguments[0];
      opts.value = arguments[1];
      callback = arguments[3];
```

I tried adding some logging here:

```js
Kv.prototype.set = function(opts, callback) {
  console.log("************************** Kv.prototype.set 1 " + arguments.length);
  switch (arguments.length) {
    case 4:
      // set(key, value, opts, callback)
      console.log("    key: " + arguments[0]);
      opts = arguments[2];
      console.log("    key: " + arguments[0]);
      opts.key = arguments[0];
```

And, this prints:

```
************************** Kv.prototype.set 1 4
    key: staging/loop/version
    key: [object Object]
```

The problem here is that this line:

```js
opts = arguments[2];
```

modifies `opts`, but `opts` is a named variable in the argument list, so this *also* modifies `arguments[0]`.  As a simplified minimal example:

```js
function test(a, b) {
    console.log(arguments[0]);
    a = arguments[1]; // Assigning `a` changes `arguments[0]`.
    console.log(arguments[0]);
}

test(1,2);
```

This prints:

```
1
2
```

(The weird part is, I've been calling this using the four-argument variant for years now, and then just today it stopped working.  I have no idea why.  This is how `arguments` has behaved in node.js for a long time, though.)

**Comment by jwalton** [comment]
(props to @kreisys who figured this out.)

**Comment by jwalton** [comment]
Thanks for the prompt fix, sir.

**Comment by silas** [comment]
Thanks for the detailed bug report! That behavior is definitely interesting... will avoid any type of argument reassignment in javascript going forward.

Fixed pushed to npm in `consul@0.34.1`.
